plugins/outputs/splunk/splunk.go

```diff
@@ -0,0 +1,298 @@
+	}
+	return nil
+}
+
+func (s *Splunk) SetSerializer(serializer serializers.Serializer) {
+	s.serializer = serializer
+}
+
+func (s *Splunk) Write(measures []telegraf.Metric) error {
+	//  If there are no metrics, just stop now.
+	if len(measures) == 0 {
+		return nil
+	}
+
+	var allMetrics = []SplunkMetric{}
+	splunkMetric := SplunkMetric{}
+
+	// -----------------------------------------------------------------------------------------------------------------------
+	//  Loop through each measure (input plugin)
+	// -----------------------------------------------------------------------------------------------------------------------
+	for _, m := range measures {
+		//fmt.Printf("\n\n Processing %s \n  %+v \n\n",m.Name(), m.Fields())
+
+		// -----------------------------------------------------------------------------------------------------------------------
+		//  Loop through tags for each measure    (add ALL Tags (except host) to each "field" element)
+		// -----------------------------------------------------------------------------------------------------------------------
+		fields := make(map[string]interface{})     // convert the Tags array from map[string]string to map[string]interface{}
+		for tagName, value := range m.Tags() {
+			//fmt.Printf(" tag values:  %s:%s \n",tagName,value)
+			// if the tagName == 'host', set the Splunk metric 'header' value == host
+			if tagName == "host" {
+				splunkMetric.Host = value
+			}else {
+				fields[tagName] = value
+			}
+		}
+
+		// -----------------------------------------------------------------------------------------------------------------------
+		//  Loop through each metric and create a map of all tags + the name and value of the metric  (Splunk Metric format)
+		// -----------------------------------------------------------------------------------------------------------------------
+		for fieldName, value := range m.Fields() {
+			fields["metric_name"] = processFieldName(m.Name(), fieldName, s)
+
+			metricValue, buildError := processFieldValue(value, fieldName, s)
+			if buildError != nil {
+				log.Printf("D! [Splunk] %s\n", buildError.Error())
+				continue
+			}
+			fields["_value"] = metricValue
+
+			splunkMetric.Time = m.Time().UnixNano()  // convert metric nanoseconds to unix time
+			splunkMetric.Event = "metric"
+			splunkMetric.Fields = fields
+
+		}   //for fieldName, value := range m.Fields() {
+		
+		// add this metric to the splunkMetric array
+		allMetrics = append(allMetrics, splunkMetric)
+
+	}   //for _, m := range measures {
+
+
+
+	// -----------------------------------------------------------------------------------------------------------------
+	//  Now we can send this set of metrics to Splunk
+	//
+	//  Create a []byte array to send via an HTTP POST
+	// -----------------------------------------------------------------------------------------------------------------
+	var payload []byte
+	var err error
+	payload, err = json.Marshal(allMetrics)
```

Comment:
Does this work? I haven't tested with an array, but the splunk [http event collector doc](http://dev.splunk.com/view/event-collector/SP-CAAAE6P) says it wants JSON streaming format:

> Here is an example of batched data. The batch protocol for HTTP Event Collector is simply event objects stacked one after the other as shown here, and not in a JSON array. Note that these events, though they only contain the "event" and "time" keys, are still valid:
```
{
  "event":"event 1", 
  "time": 1447828325
}

{
  "event":"event 2", 
  "time": 1447828326
}
```

----

To get this format in GO, you need to use a [`json.Encoder`](https://golang.org/pkg/encoding/json/#Encoder). E.G.:

```
buf := bytes.NewBuffer(nil)
enc := json.NewEncoder(buf)
for _, m := range measures {
  ...
  enc.Encode(splunkMetric)
}
http.NewRequest("POST", s.SplunkUrl, buf)
```

Reply:
This data will be placed into a Splunk metrics index as opposed to an event index.  The format of a metric is still JSON, but different from event data.    The byte array does indeed work ;-).